src/System.Data.SqlClient/src/System.Data.SqlClient.csproj

```diff
@@ -520,6 +520,10 @@
     <Compile Include="System.Data.SqlClient.TypeForwards.cs" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(IsFunctionalAssembly)' == 'true'">
```

Comment:
Why do we need this condition?

Reply:
because SNI is a native library that's only ever used by real implementation.

Reply:
Actually, the condition needs to change, since it doesn't make its way into the package (RID specific dependencies).  I'll fix it.